### ISSUE TITLE
Code Hike components not rendering correctly inside <details> tag on Firefox

### ISSUE THREAD

**Issue by gregnr** [supabase/supabase#14850]
# Bug report

## Describe the bug

Markdown code blocks inside a [`<details>` disclosure element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/details) don't render correctly upon expanding in Firefox. Chrome works fine.

We are rendering code blocks via [Code Hike](https://codehike.org/), so the bug may have to do with the way their components render.

## To Reproduce

1. In Firefox, go to https://supabase.com/docs/guides/database/vault
2. Expand the first "Show Result" element
3. No code block appears
    ![image](https://github.com/supabase/supabase/assets/4133076/b97c9969-3d40-42a3-9816-4cc3203e9015)

**Comment by Vlad116** [comment]
Can I work on this issue? @gregnr @kiwicopple

**Comment by gregnr** [comment]
You bet @Vlad116!

**Comment by Vlad116** [comment]
> К сожалению, похоже, что это исходная проблема с Codehike, я подал туда проблему: [code-hike/codehike#372](https://github.com/code-hike/codehike/issues/372) 
You are right, without solving the problem in CodeHike (this bug is reproduced in their sandbox), this can only be solved by creating a bike. 
I was going to do the same yesterday, but you beat me to it)

**Comment by MildTomato** [comment]
would it just be easier to make our own collapsible using radix. 
we already have one as well in ui package.

**Comment by pomber** [comment]
https://github.com/code-hike/codehike/pull/373 should fix it, but it will be shipped in v0.9.0 which also includes some changes to themes and CSS that probably affect Supabase docs.

**Comment by mdjohns** [comment]
> would it just be easier to make our own collapsible using radix. we already have one as well in ui package.

Sounds like this is the easiest solution, since the 0.9.0 upgrade will probably require a bit more scrutiny to ensure there is no other impact.

@pomber 's workaround from code-hike/codehike#373:
> The workaround is to make sure that the component re-renders when the visibility change. In the `details` example that would be wrapping the code in a div, listening to the open event, and changing the key of the wrapper when the visibility changes.

This may not be necessary for a new component, since we could just conditionally render it, but thought I'd include it here to be thorough.

**Comment by mdjohns** [comment]
Looks like `0.9.0` [has been released](https://github.com/code-hike/codehike/releases/tag/v0.9.0)!

Should we try an upgrade @gregnr? This may be something @Vlad116 could still take on if so.

**Comment by gregnr** [comment]
Let's do it! Just a note, we had to override the `@code-hike/lighter` version to get access to the `from-css` logic before it was officially supported:
https://github.com/supabase/supabase/blob/ef8769638ca371eaafb8b1f0897599d7e281b82d/package.json#L36-L40

We should remove this as part of the upgrade.

Code Hike theme lives here:
https://github.com/supabase/supabase/blob/ef8769638ca371eaafb8b1f0897599d7e281b82d/packages/config/code-hike.theme.json

You will need to verify if there were any breaking changes that we need to fix:
https://github.com/code-hike/codehike/pull/370

Good luck!

**Comment by mdjohns** [comment]
Hey all, 

Took a quick stab to see how it would go simply updating the `@code-hike/mdx` version and removing the `@code-hike/lighter` override above.

It looks like the CSS [here](https://github.com/supabase/supabase/blob/master/packages/config/code-hike.scss) may need some tweaking. 

Unfortunately, CSS is a bit of a weak spot for me. Any tips?

<details>
  <summary>Screenshot (Chrome 113 on PopOS 22.04)</summary>

![image](https://github.com/supabase/supabase/assets/13648640/2b765cbf-14a3-4de2-9214-c38d5947f46c)

</details>

**Comment by pomber** [comment]
(Disclaimer: I haven't looked at the full setup on this website) What I recommend is to import the CSS from `@code-hike/mdx/styles` and then apply all the customizations from Supabase. That way is easier to upgrade. Also, start with one of the [built-in themes](https://codehike.org/docs/themes) instead of the custom one until you fixed the CSS, then you can update the theme.
